torchvision/utils.py

```diff
@@ -191,34 +192,42 @@ def draw_bounding_boxes(
     elif image.size(0) not in {1, 3}:
         raise ValueError("Only grayscale and RGB images are supported")
 
+    if labels:
+        if len(labels) != boxes.size(0):
+            raise ValueError("Number of boxes and labels mismatch. Please specify labels for each box.")
+
+    # List of colors should have minimum colors.
+    if colors:
+        if isinstance(colors, list):
+            if len(colors) < boxes.size(0):
+                raise ValueError("Number of colors should be greater or equal to the number of boxes.")
```

Comment:
Nit: same here we could indicate the exact values.

Also, should we raise an error if num_colors != num_boxes in general?

Reply:
Nope, I would disagree. Providing excess colors makes it easy to define a larger color palette. Like say of 12 colors and utility would dynamically choose colors as needed. This is essentially a feature and intended :smiley: